src/test/java/io/cdap/plugin/sfmc/sink/DataExtensionClientTest.java

```diff
@@ -0,0 +1,220 @@
+    Schema schema = Schema.recordOf("record",
+      Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("markedPrice", Schema.nullableOf(Schema.decimalOf(4, 2))),
+      Schema.Field.of("updated", Schema.of(Schema.Type.BOOLEAN)),
+      Schema.Field.of("totalProducts", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("dt", Schema.nullableOf(Schema.of(Schema.LogicalType.DATE))),
+      Schema.Field.of("bytedata", Schema.of(Schema.Type.BYTES)),
+      Schema.Field.of("null", Schema.of(Schema.Type.NULL)),
+      Schema.Field.of("timestamp",
+        Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))));
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    Collection<ETDataExtensionColumn> columns = new ArrayList<>();
+    ETDataExtensionColumn column1 = new ETDataExtensionColumn();
+    column1.setId("121");
+    column1.setDescription("newCustomer");
+    column1.setLength(22);
+    column1.setName("name");
+    ETDataExtensionColumn column2 = new ETDataExtensionColumn();
+    Date d = new Date(2022, 02, 9);
+    column2.setCreatedDate(d);
+    column2.setDefaultValue("default");
+    column2.setName("dt");
+    column2.setType(ETDataExtensionColumn.Type.DECIMAL);
+    ETDataExtensionColumn column3 = new ETDataExtensionColumn();
+    column3.setIsPrimaryKey(true);
+    column3.setIsRequired(true);
+    column3.setKey("key");
+    column3.setName("updated");
+    column3.setType(ETDataExtensionColumn.Type.NUMBER);
+    ETDataExtensionColumn column4 = new ETDataExtensionColumn();
+    column4.setModifiedDate(d);
+    column4.setName("id");
+    column4.setType(ETDataExtensionColumn.Type.LOCALE);
+    ETDataExtensionColumn column6 = new ETDataExtensionColumn();
+    column6.setModifiedDate(d);
+    column6.setName("null");
+    column6.setType(ETDataExtensionColumn.Type.BOOLEAN);
+    ETDataExtensionColumn column7 = new ETDataExtensionColumn();
+    column7.setModifiedDate(d);
+    column7.setName("markedPrice");
+    column7.setType(ETDataExtensionColumn.Type.DATE);
+    columnList.add(column1);
+    columnList.add(column2);
+    columnList.add(column3);
+    columnList.add(column4);
+    columnList.add(column6);
+    columnList.add(column7);
+    columns.add(column7);
+    columns.add(column1);
+    columns.add(column2);
+    columns.add(column3);
+    columns.add(column4);
+    columns.add(column6);
+    String dataExtensionKey = "DE";
+    PowerMockito.mockStatic(ETDataExtension.class);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(getClient(), dataExtensionKey));
+    PowerMockito.when(ETDataExtension.retrieveColumns(Mockito.any(), Mockito.anyString())).thenReturn(columnList);
+    try {
+      dataExtensionClient.validateSchemaCompatibility(schema, collector);
+      collector.getOrThrowException();
+
```

Comment:
Fail the test if exception is not thrown as expected. Similar comment in other places